main.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const API_URL = 'https://opentdb.com/api.php?amount=10';
+const title = document.getElementById('title');
+const msgWindow = document.getElementById('msg-window');
+const quizInfo = document.getElementById('quiz-info');
+const janre = document.getElementById('janre');
+const difficulty = document.getElementById('difficulty');
+const choices = document.getElementById('choices');
+const restart = document.getElementById('restart');
+let quizData = '';
+let correctCnt = 0;   // 正解数 
+let i = 0;            // クイズデータ配列のインデックス
+
+document.getElementById('start').addEventListener('click', function() {
+  title.textContent = '取得中';
+  msgWindow.textContent = '少々お待ちください';
+  document.getElementById('start').classList.add('hidden');
+  
+  fetch(API_URL)
+    .then(response => {
+      if (!response.ok) throw new Error('データの取得に失敗しました');
+      return response.json();
+    })
+    .then(data => {
+      quizData = data.results;
+      createQuiz(i);
+    })
+    .catch(e => {
+      title.textContent = 'エラー';
+      msgWindow.textContent = e.message;
+    });
+});
```

Comment:
アロー関数を使う + `async` `await` を使う方がイマドキかなと思いました。
他言語と見た目も近しくなるので、見る人にも優しいです。

```suggestion
document.getElementById('start').addEventListener('click', async () => {
  title.textContent = '取得中';
  msgWindow.textContent = '少々お待ちください';
  document.getElementById('start').classList.add('hidden');
  try {
    const response = await fetch(API_URL);
    if (!response.ok) throw new Error('データの取得に失敗しました');
    const data = await response.json();
    quizData = data.results;
    createQuiz(i);
  } catch (e) {
    title.textContent = 'エラー';
    msgWindow.textContent = e.message;    
  }
})
```